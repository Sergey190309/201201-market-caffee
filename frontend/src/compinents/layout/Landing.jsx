import React from 'react'
import {
  Segment,
  Container,
  Header
} from 'semantic-ui-react'

const Landing = () => (
    <Container>
      <Segment color='violet'>
        <Header
        textAlign='center'
        size='medium'
        color='yellow'>
          That's header
        </Header>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Atque quis fugiat delectus repellat iste, nisi, temporibus ullam impedit officia nulla reprehenderit quas cupiditate libero sapiente enim illo veniam harum velit.
        </p>
        <p>
          Lorem ipsum dolor sit amet consectetur, adipisicing elit. Dolore necessitatibus maxime veritatis expedita quos fuga repudiandae. Eligendi, doloremque fugiat, similique, repellendus dolores velit magni ab quibusdam nesciunt eos accusantium reiciendis!
        </p>
      </Segment>
    </Container>
  )

export default Landing
