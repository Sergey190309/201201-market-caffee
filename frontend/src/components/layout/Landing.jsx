import React from "react";
import { Segment, Container, Header } from "semantic-ui-react";

const Landing = () => (
  <Container>
    <Segment color="violet">
      <Header textAlign="center" size="medium" color="yellow">
        That's header
      </Header>
      <p>Lorem ipsum dolor sit, amet consectetur adipisicing elit. Iusto, provident ratione numquam praesentium dolorem nostrum dolore asperiores in error dolores nihil quidem reprehenderit ut id maiores deleniti officiis eaque voluptates.</p>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Atque quis
        fugiat delectus repellat iste, nisi, temporibus ullam impedit officia
        nulla reprehenderit quas cupiditate libero sapiente enim illo veniam
        harum velit.
      </p>
      <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Quibusdam facere possimus eligendi delectus recusandae a quaerat, at officiis! Culpa odit sed adipisci, saepe eum repellat distinctio cupiditate tempora magni totam!</p>
    </Segment>
  </Container>
);

export default Landing;
