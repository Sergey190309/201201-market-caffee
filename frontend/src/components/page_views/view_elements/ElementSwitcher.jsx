import React, { Fragment } from 'react';
import { Divider } from 'semantic-ui-react';
import ViewHeader from './ViewHeader';
import ViewFooter from './ViewFooter';
import ViewVBlock from './ViewVBlock';
import ViewHBlock from './ViewHBlock';

export const Output = ({ structure, viewName }) => {
  // console.log(structure);
  const keys = Object.keys(structure);
  const output = keys.map((key, index) => {
    const componentType = structure[key]['type'];
    const componentSubType = structure[key]['subtype'] ? structure[key]['subtype'] : null;
    const subComponentQnt = structure[key]['qnt'] ? structure[key]['qnt'] : null;
    let component;
    const recordId =
      `${key}_${componentType}` +
      (componentSubType ? `_${componentSubType}` : '') +
      (subComponentQnt ? `_${subComponentQnt}` : '');
    // console.log(recordId)
    const props = { recordId: recordId, viewName: viewName };
    switch (componentType) {
      case 'header':
        component = <ViewHeader {...props} />;
        break;
      case 'footer':
        component = <ViewFooter {...props} />;
        break;
      case 'hblock':
        component = <ViewHBlock {...props} />;
        break;
      case 'vblock':
        component = <ViewVBlock {...props} />;
        break;
      default:
        component = <ViewHeader />;
    }
    return (
      <Fragment key={key}>
        {component}
        {index < keys.length - 1 ? <Divider /> : null}
      </Fragment>
    );
  });
  // console.log('output, output ->', output);
  return output;
};

export default Output;
