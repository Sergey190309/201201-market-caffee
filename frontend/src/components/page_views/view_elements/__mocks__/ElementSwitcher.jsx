import React from 'react';

const ElementSwitcher = ({ structure, viewName, lng }) => {
  // useEffect(() => {
  //   // console.log('useEffect')
  // })

  const keys = Object.keys(structure)
  // console.log('ElementSwitcher, keys ->', keys)
  const _output = keys.map((key, index) => {
    const _item = Object.entries(structure[key])
    // console.log('mockElementSwitcher, structure ->', structure[key])
    return (
      <div key={key} data-testid={key}>
        {_item}
      </div>
    )
  })
  return (
    <div data-testid='ElementSwitcher'>
      ElementSwitcher
      {_output}
      {viewName}
      {lng}
    </div>
  )
};

export default ElementSwitcher;
