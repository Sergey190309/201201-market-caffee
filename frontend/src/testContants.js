export const landingPageStructure = {
  '00': { type: 'header' },
  '01': { qnt: 3, type: 'vblock', subtype: 'txt' },
  '02': { qnt: 2, type: 'hblock', subtype: 'pix' },
  '03': { qnt: 2, type: 'vblock', subtype: 'pix' },
  '04': { type: 'footer' },
}
