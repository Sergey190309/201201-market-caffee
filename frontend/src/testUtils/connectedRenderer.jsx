import React from 'react';

import { BrowserRouter } from 'react-router-dom';

import { render } from '@testing-library/react';

import { createStore } from 'redux';
import { Provider } from 'react-redux';

import rootReducer from '../redux/reducers';

const connectedRender = (
  ui,
  { initialState, store = createStore(rootReducer, initialState), ...renderOptions } = {}
) => {
  const Wrapper = ({ children }) => {
    return (
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  };
  return render(ui, { wrapper: Wrapper, ...renderOptions });
};

// const linkedRender = (ui) => {}

export * from '@testing-library/react';
export { connectedRender };
