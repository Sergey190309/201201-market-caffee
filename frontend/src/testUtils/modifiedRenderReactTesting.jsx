import React from 'react';

import { BrowserRouter } from 'react-router-dom';

import { render } from '@testing-library/react';

import { createStore, combineReducers } from 'redux';
import { Provider } from 'react-redux';

import alerts from '../redux/reducers/alert'
import device from '../redux/reducers/device'
import logIn from '../redux/reducers/auth'
import lng from '../redux/reducers/lng'

// import rootReducer from '../redux/reducers';
const rootReducer = combineReducers({
  alerts,
  device,
  logIn,
  lng
});

const connectedRender = (
  ui,
  { initialState, store = createStore(rootReducer, initialState), ...renderOptions } = {}
) => {
  const Wrapper = ({ children }) => {
    return <Provider store={store}>{children}</Provider>;
  };
  return render(ui, { wrapper: Wrapper, ...renderOptions });
};

const linkedRender = (ui, { ...renderOptions } = {}) => {
  const Wrapper = ({ children }) => {
    return <BrowserRouter>{children}</BrowserRouter>;
  };
  return render(ui, { wrapper: Wrapper, ...renderOptions });
};

const connectedLinkedRender = (
  ui,
  { initialState, store = createStore(rootReducer, initialState), ...renderOptions } = {}
) => {
  const Wrapper = ({ children }) => {
    return (
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  };
  return render(ui, { wrapper: Wrapper, ...renderOptions });
};

// const linkedRender = (ui) => {}

export * from '@testing-library/react';
export { connectedRender, linkedRender, connectedLinkedRender };
