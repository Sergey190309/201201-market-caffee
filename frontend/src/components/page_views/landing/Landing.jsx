import isEmpty from 'lodash/isEmpty';
import React, { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import { Segment, Container } from 'semantic-ui-react';

import ElementSwitcher from '../view_elements/ElementSwitcher';
import { lngSelector, structureSelector } from '../../../redux/slices';

import { viewSegmentColor } from '../../../utils/colors';

export const getLoadedStructure = (pageName, structures) => {
  /**
   * The function created for testing.
   * Reciev all structures, return one that corresspong to component name
   */
  const { [pageName]: value } = structures;
  return value;
};

export const Landing = ({ getLoadedStructure }) => {
  const [structure, setStructure] = useState({});
  const [componentName] = useState('landing');
  const { lng } = useSelector(lngSelector);
  const { loaded } = useSelector(structureSelector);

  const loadedStructures = useSelector(structureSelector);

  useEffect(() => {
    if (loaded) {
      setStructure(getLoadedStructure(componentName, loadedStructures));
    }
  }, [loaded, getLoadedStructure, componentName, loadedStructures]);

  const _output = structure => {
    return <ElementSwitcher viewName={componentName} structure={structure} lng={lng} />;
  };
  // console.log('Landing, loadedStructure ->', loadedStructure)

  return (
    <Container data-testid='LandingContainer'>
      <Segment color={viewSegmentColor} data-testid='LandingSegment'>
        {/* {_output(structure)} */}
        {isEmpty(structure) ? null : _output(structure)}
      </Segment>
    </Container>
  );
};

Landing.defaultProps = {
  getLoadedStructure: getLoadedStructure,
};

Landing.propTypes = {
  getLoadedStructure: PropTypes.func.isRequired,
};

export default Landing;
