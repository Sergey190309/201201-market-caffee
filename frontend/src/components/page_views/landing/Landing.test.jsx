import React from 'react';
// import ShallowRenderer from 'react-test-renderer/shallow';
import { render, screen } from '../../../testUtils';
// import '@testing-library/jest-dom';
// import userEvent from '@testing-library/user-event';

import { Landing } from './Landing';
import { landingPageStructure } from '../../../testContants';
import ElementSwitcher from '../view_elements/ElementSwitcher';

const mockElementSwitcher = ({ structure, viewName, lng }) => {
  return <div data-testid='ElementSwitcher' />;
};

jest.mock('../view_elements/ElementSwitcher', () => ({
  __esModule: true,
  default: jest.fn(),
}));

describe('Landing page testing', () => {
  beforeEach(() => {
    ElementSwitcher.mockImplementation(mockElementSwitcher);
  });

  const testProps = {
    structureLoaded: true,
    loadedStructure: landingPageStructure,
    lng: 'en',
  };

  test('it should exist and rendered (snapshot)', () => {
    const result = render(<Landing {...testProps} />);
    expect(result).toMatchSnapshot();
  });
  describe('appearance, rendering child with proper props', () => {
    // const keys = Object.keys(testProps['loadedStructure']);

    test('rendering with props (structure)', () => {
      const { loadedStructure, lng } = testProps
      const expArgs = {structure: loadedStructure, lng, viewName: 'landing'}
      render(<Landing {...testProps} />);
      expect(ElementSwitcher).toHaveBeenCalledTimes(1);
      // console.log('Landing page testing, testProps ->', testProps);
      expect(ElementSwitcher.mock.calls[0][0]).toEqual(expArgs);

      const LandingSegment = screen.getByTestId('LandingSegment');
      expect(LandingSegment).not.toBeEmptyDOMElement()
      expect(LandingSegment).toHaveClass('segment');

      // screen.debug();
    });

    test('rendering witout props (structure)', () => {
      render(<Landing />);
      const landingContainer = screen.getByTestId('LandingContainer');
      expect(landingContainer).toHaveClass('container');

      const LandingSegment = screen.getByTestId('LandingSegment');
      expect(LandingSegment).toBeEmptyDOMElement();
      expect(LandingSegment).toHaveClass('segment');
      // console.log('landing page testing, LandingSegment ->', LandingSegment)
    });
  });
});
