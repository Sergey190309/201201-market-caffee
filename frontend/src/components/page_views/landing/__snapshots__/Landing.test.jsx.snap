// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page testing it should exist and rendered (snapshot) 1`] = `
<div>
  <div
    class="ui container"
    data-testid="LandingContainer"
  >
    <div
      class="ui violet segment"
      data-testid="LandingSegment"
    >
      <div
        data-testid="ElementSwitcher"
      >
        ElementSwitcher
        {"00":{"type":"header"},"01":{"qnt":3,"type":"vblock","subtype":"txt"},"02":{"qnt":2,"type":"hblock","subtype":"pix"},"03":{"qnt":2,"type":"vblock","subtype":"pix"},"04":{"type":"footer"}}
        landing
        en
      </div>
    </div>
  </div>
</div>
`;
